syntax = "proto3";
package translation_service;

import "google/protobuf/empty.proto";

message TranslationConfig {
  string src_lang = 1;
  string tgt_lang = 2;
}

message TranslationRequest {
  TranslationConfig config = 1;
  string text = 2;
}

message TranslationStreamRequest {
  oneof request {
    TranslationConfig config = 1;
    string text = 2;
  }
}

message SupportedLanguagesResponse {
  repeated TranslationConfig models = 1;
}

message TranslationResponse {
  string text = 1;
}

service TranslationService {
  rpc Ping(google.protobuf.Empty) returns (google.protobuf.Empty) {}
  rpc Shutdown(google.protobuf.Empty) returns (google.protobuf.Empty) {}
  rpc SupportedLanguages(google.protobuf.Empty) returns (SupportedLanguagesResponse) {}
  rpc Translate(TranslationRequest) returns (TranslationResponse) {}
  rpc TranslateStream(stream TranslationStreamRequest) returns (stream TranslationResponse) {}
}
